st7920.py

```diff
@@ -128,7 +130,17 @@ def put_text(self, s, x, y):
 				pass
 			x += cw
 	
-	def redraw(self, dx1=0, dy1=0, dx2=127, dy2=63):
-		for i in range(dy1, dy2+1):
-			self.send(0,0,[0x80 + i%32, 0x80 + ((dx1/16) + (8 if i>=32 else 0))]) # set address
-			self.send(1,0,self.fbuff[i][dx1/8:(dx2/8)+1])
+	def redraw(self, dx1=0, dy1=0, dx2=127, dy2=63, full=False):
+		if full or self.currentlydisplayedfbuff == None:
+			for i in range(dy1, dy2+1):
+				self.send(0,0,[0x80 + i%32, 0x80 + ((dx1//16) + (8 if i>=32 else 0))]) # set address
+				self.send(1,0,self.fbuff[i][dx1//8:(dx2//8)+1])
+		else:
+			lines = []
+			for i in range(len(self.fbuff)):

```

Comment:
This range should be over `dy1, dy2+1`, so the behaviour is the same as when performing a 'full' redraw: only the lines which changed _within the range specified_ are sent to the screen.